core/src/stages/move_to.cpp

```diff
@@ -204,47 +202,81 @@ bool MoveTo::compute(const InterfaceState& state, planning_scene::PlanningSceneP
 		// plan to joint-space target
 		success = planner_->plan(state.scene(), scene, jmg, timeout, robot_trajectory, path_constraints);
 	} else {  // Cartesian goal
-		const moveit::core::LinkModel* link;
-		Eigen::Isometry3d target_eigen;
-
-		// Cartesian targets require an IK reference frame
+		// Where to go?
+		Eigen::Isometry3d target;
+		// What frame+offset in the robot should go there?
 		geometry_msgs::PoseStamped ik_pose_msg;
+
 		const boost::any& value = props.get("ik_frame");
+
+		auto get_tip{ [&jmg](std::string& out) {
+			// determine IK frame from group
+			std::vector<std::string> tips;
+			jmg->getEndEffectorTips(tips);
+			if (tips.size() != 1) {
+				return false;
+			}
+			out = tips[0];
+			return true;
+		} };
+
 		if (value.empty()) {  // property undefined
-			// determine IK link from group
-			if (!(link = jmg->getOnlyOneEndEffectorTip())) {
+			if (!get_tip(ik_pose_msg.header.frame_id)) {
 				solution.markAsFailure("missing ik_frame");
 				return false;
 			}
-			ik_pose_msg.header.frame_id = link->getName();
 			ik_pose_msg.pose.orientation.w = 1.0;
 		} else {
 			ik_pose_msg = boost::any_cast<geometry_msgs::PoseStamped>(value);
-			if (!(link = robot_model->getLinkModel(ik_pose_msg.header.frame_id))) {
-				solution.markAsFailure("unknown link for ik_frame: " + ik_pose_msg.header.frame_id);
+			if (ik_pose_msg.header.frame_id.empty() && !get_tip(ik_pose_msg.header.frame_id)) {
+				solution.markAsFailure("frame_id of ik_frame is empty and no unique group tip was found");
+				return false;
+			} else if (!scene->knowsFrameTransform(ik_pose_msg.header.frame_id)) {
+				std::stringstream ss;
+				ss << "ik_frame specified in unknown frame '" << ik_pose_msg << "'";
+				solution.markAsFailure(ss.str());
 				return false;
 			}
 		}
```

Comment:
This block of code is required 1:1 for `MoveRelative` as well:
https://github.com/ros-planning/moveit_task_constructor/blob/7ec874572c3b35f1cdd7a71c5e6b94c749b3093b/core/src/stages/move_relative.cpp#L199-L213

We should factor that out into a function...

Reply:
please have another look. I'm happy with the code again (assuming CI is) and I addressed this too.